### PR TITLE
Move summary function to trait, reduce testbase redundancy, cleanup

### PR DESCRIPTION
**Pull request by tassilogroeper**
worked with Martin Taleski on this. Link https://www.drupal.org/node/2456849

### CODE REVIEW

File: src/Core/RulesPluginTrait.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Component\Plugin\Exception\ContextException;
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait RulesPluginTrait {
```
Comment (fago) [inline src/Core/RulesPluginTrait.php]: There are RulesPlugins which do not need this trait - thus the naming needs work. I'd suggest doing an ExecutablePluginTrait - as those methods stem from there.

File: src/Core/RulesPluginTrait.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  private function getLabelValue() {
+    $definition = $this->getPluginDefinition();
```
Comment (fago) [inline src/Core/RulesPluginTrait.php]: I don't see why the contents of this method isn't just in the summary() method?

File: src/Core/RulesPluginTrait.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    return $definition['label'];
+  }
+
+  /**
+   * Get the translated summary from the label annotation
+   *
+   * @return string
+   * @throws \Drupal\Component\Plugin\Exception\ContextException
```
Comment (fago) [inline src/Core/RulesPluginTrait.php]: Check drupal coding styles for correct ordering and missing descriptions.

File: src/Core/RulesPluginTrait.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    if (empty($definition['label'])) {
+      throw new ContextException('The label is not defined.');
+    }
+
+    return $definition['label'];
+  }
+
+  /**
+   * Get the translated summary from the label annotation
```
Comment (fago) [inline src/Core/RulesPluginTrait.php]: Missing trailing point. But should say it's "Implements FQClassname::summary()."

File: tests/src/Unit/RulesActionBaseTest.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?php
```
Comment (fago) [inline tests/src/Unit/RulesActionBaseTest.php:1]: Missing NL after it

File: tests/src/Unit/RulesActionBaseTest.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * @coversDefaultClass \Drupal\rules\Core\RulesActionBase
+ * @group rules
+ */
+class RulesActionBaseTest extends RulesUnitTestBase {
+
+  /**
+   * Test the summary is being parsed from the label annotation
+   *
+   * @expectedException     \Drupal\Component\Plugin\Exception\ContextException
```
Comment (fago) [inline tests/src/Unit/RulesActionBaseTest.php]: Too many spaces in between

File: tests/src/Unit/RulesActionBaseTest.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+namespace Drupal\Tests\rules\Unit;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Core\RulesActionBase
+ * @group rules
+ */
+class RulesActionBaseTest extends RulesUnitTestBase {
+
+  /**
+   * Test the summary is being parsed from the label annotation
```
Comment (fago) [inline tests/src/Unit/RulesActionBaseTest.php]: Missing trailing point. Docu leaves unclear why this throws an exception.

File: tests/src/Unit/RulesActionBaseTest.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+   * @expectedException     \Drupal\Component\Plugin\Exception\ContextException
+   * @covers ::summary
+   */
+  public function testSummaryThrowingException() {
+    $rulesActionBase = $this->getMockForAbstractClass('Drupal\rules\Core\RulesActionBase', [[], '', '']);
+    $rulesActionBase->summary();
+  }
+
+  /**
+   * Test the summary is being parsed from the label annotation
```
Comment (fago) [inline tests/src/Unit/RulesActionBaseTest.php]: Missing trailing point.

File: tests/src/Unit/RulesConditionBaseTest.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+namespace Drupal\Tests\rules\Unit;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Core\RulesConditionBase
+ * @group rules
+ */
+class RulesConditionBaseTest extends RulesUnitTestBase {
+
+  /**
+   * Test the summary is being parsed from the label annotation
```
Comment (fago) [inline tests/src/Unit/RulesConditionBaseTest.php]: See comments as in action base test.

File: src/Core/RulesPluginTrait.php (hunk 2 above)
Comment (tassilogroeper) [inline src/Core/RulesPluginTrait.php]: Due to naming confusion I preferred to get the label first and then make it accessible through the public function. The annotation sets the label, but the Plugin (as well as Plugin/Condition/Language::summary) kinda uses this summary function. I thought, that would be a more consistent way.

But feel free to rename it.

File: src/Core/RulesPluginTrait.php (hunk 4 above)
Comment (tassilogroeper) [inline src/Core/RulesPluginTrait.php]: I can't think of any "implements" here. what do you mean by that?

**Comment by tassilogroeper** [comment]
Did the renaming and monks ;)

**Comment by tassilogroeper** [comment]
travis failed due to connection problem to github...

File: src/Core/ExecutablePluginTrait.php [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+trait ExecutablePluginTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  private function getLabelValue() {
+    $definition = $this->getPluginDefinition();
+
+    if (empty($definition['label'])) {
+      throw new ContextException('The label is not defined.');
```
Comment (klausi) [inline src/Core/ExecutablePluginTrait.php:24]: Context exception? This has nothing to do with the context, maybe a plugin exception would be better?

File: src/Core/ExecutablePluginTrait.php [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+   *
+   * @throws \Drupal\Component\Plugin\Exception\ContextException
+   *  Thrown if a summary was not set.
+   *
+   * @return string
+   */
+  public function summary() {
+    return $this->getLabelValue();
+  }
+}
```
Comment (klausi) [inline src/Core/ExecutablePluginTrait.php:41]: missing newline

**Comment by klausi** [comment]
Continued in #180

**Pull request closed without merging**